crates/ethereum-rpc/src/subscription.rs

```diff
@@ -125,12 +120,43 @@ pub async fn handle_logs_subscription(
                         &log,
                     )
                     .unwrap();
-                    let Ok(_) = subscription.send(msg).await else {
-                        // Connection closed
-                        return;
-                    };
+                    send_tasks.push(subscription.send(msg));
                 }
             }
         }
-    });
+        let _ = future::join_all(send_tasks).await;
+        // Drop lock to release the read lock.
+        drop(subscriptions);
```

Comment:
I am not sure if this is necessary. I think subscriptions is dropped after each iteration automatically.

Reply:
IMO, it's better to be explicit about it. Since `RwLock` would be waited upon elsewhere for write which means that we want to make sure we don't starve the write task. You're right about it being dropped but again i want to be explicit.